modules/solid_mechanics/test/tests/neml2/interpolated_material/moose_neml2_return_mapping.i

```diff
@@ -0,0 +1,183 @@
+# NEML2 file in MPA
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 3
+    nx = 5
+    ny = 5
+    nz = 5
+  []
+[]
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Physics]
+  [SolidMechanics]
+    [QuasiStatic]
+      [all]
+        strain = SMALL
+        new_system = true
+        add_variables = true
+        formulation = TOTAL
+        volumetric_locking_correction = true
+      []
+    []
+  []
+[]
+
+[BCs]
+  [xfix]
+    type = DirichletBC
+    variable = disp_x
+    boundary = left
+    value = 0
+  []
+  [yfix]
+    type = DirichletBC
+    variable = disp_y
+    boundary = bottom
+    value = 0
+  []
+  [zfix]
+    type = DirichletBC
+    variable = disp_z
+    boundary = back
+    value = 0
+  []
+  [pressure_x]
+    type = Pressure
+    variable = disp_x
+    boundary = right
+    function = pressure_fcn
+  []
+[]
+
+# -- interpolation grids in  'models/random_value_grid.json'
+# "in_stress": [0.0, 30, 60, 100, 130, 160, 200, 230, 260, 320.0],
+# "in_temperature": [600, 650, 750, 800, 850, 900, 950, 1000, 1100, 1200.0],
+# "in_plastic_strain": [0.0, 0.0001, 0.001, 0.01],
+# "in_cell": [1000000.0, 10000000000000.0],
+# "in_wall": [1000000000000.0, 10000000000000.0],
+# "in_env": [1e-09, 1e-06],
+
+[Functions]
+  [pressure_fcn]
+    type = ParsedFunction
+    expression = 200 #MPa
+  []
+[]
+# [AuxVariables]
+#   [T]
+#     initial_condition = 750 #K
+#   []
+# []
+[Materials]
+  [init_dd]
+    type = GenericConstantMaterial
+    prop_names = 'T init_cell_dd init_wall_dd init_envFac'
+    prop_values = '750 1e11 8e12 2e-8'
+  []
+[]
+
+[NEML2]
+  input = 'models/interp_matl_radial_return.i'
+  [all]
+    model = 'model'
+    verbose = true
+    device = 'cpu'
+
+    moose_input_types = 'MATERIAL MATERIAL MATERIAL MATERIAL MATERIAL MATERIAL MATERIAL POSTPROCESSOR POSTPROCESSOR'
```

Comment:
This is not set up correctly because I am getting the error:
```The NEML2 model requires a stateful input variable `old_state/cell_dd`, but its state counterpart on the output axis has not been retrieved by any object. Therefore, there is no way to properly propagate the corresponding stateful data in time. The common solution to this problem is to add a NEML2ToMOOSE retriever such as those called `NEML2To*MOOSEMaterialProperty`.```

Reply:
I think you need to save the corresponding neml2 output variable into moose. The output variable should be `state/cell_dd`.

Reply:
Gary gave me a fix for the batch size problem I had.  The fix goes in NEML2.  It is from linear algebra library changing between libtorch versions.  I am using pytorch version 2.1.2.post4 and I need this fix.  Gary is using 2.5.1 and does not need the fix.